lib/3scale/api.rb

```diff
@@ -5,9 +5,11 @@ module API
     autoload :Client, '3scale/api/client'
     autoload :HttpClient, '3scale/api/http_client'
 
-    def self.new(endpoint:, provider_key:)
+    def self.new(endpoint:, provider_key: nil, access_token: nil)
+      raise ArgumentError.new("access_token or provider_key should be provided") unless provider_key || access_token
```

Comment:
CodeClimate would says prefer `raise class, argument` over initialising it.

And I'd rather unify those two params before passing them along like:

```ruby
authentication = provider_key || access_token

raise ArgumentError, "access_token or provider_key missing" unless authentication

http_client = HttpClient.new(endpoint: endpoint, authentication: authentication)
```